pkg/manifests/manifests_test.go

```diff
@@ -1599,6 +1601,131 @@ alertmanagerMain:
 	}
 }
 
+func TestK8sPrometheusAdapterAuditLog(t *testing.T) {
+	// setting audit level should add the audit args
+	tt := []struct {
+		config       string
+		expectedArgs []string
+		err          error
+	}{{
+		config: ``,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/metadata-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+	}, {
+		config: `k8sPrometheusAdapter: `,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/metadata-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+	}, {
+		config: `
+k8sPrometheusAdapter:
+  audit: {} `,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/metadata-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+	}, {
+		config: `
+k8sPrometheusAdapter:
+  audit:
+    profile: Request
+`,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/request-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+		err: nil,
+	}, {
+		config: `
+k8sPrometheusAdapter:
+  audit:
+    profile: RequestResponse
+`,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/requestresponse-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+		err: nil,
+	}, {
+		config: `
+  k8sPrometheusAdapter:
+    audit:
+     profile: None
+`,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/none-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+		err: nil,
+	}, {
+		config: `
+  k8sPrometheusAdapter:
+    audit:
+      profile: Foobar  # should generate an error
+`,
+		err: ErrConfigValidation,
+	}}
+
+	for _, test := range tt {
+		c, err := NewConfigFromString(test.config)
+		if err != nil {
+			t.Logf("%s\n\n", test.config)
+			t.Fatal(err)
+		}
+		f := NewFactory("openshift-monitoring", "openshift-user-workload-monitoring", c,
+			defaultInfrastructureReader(), &fakeProxyReader{}, NewAssets(assetsPath))
+
+		d, err := f.PrometheusAdapterDeployment("foo", map[string]string{
+			"requestheader-allowed-names":        "",
+			"requestheader-extra-headers-prefix": "",
+			"requestheader-group-headers":        "",
+			"requestheader-username-headers":     "",
```

Comment:
I don't think we will need these headers for this particular test.

Reply:
Initial I did try this but I got an error
```go
  d, err := f.PrometheusAdapterDeployment("foo", map[string]string{})  // or nil
```

```
   manifests_test.go:1682: Expected error "invalid value for config" but got "value not found in extension api server authentication configmap: key requestheader-allowed-names is missing"
-
```